test/avro_schema_test.cc

```diff
@@ -47,8 +47,56 @@ void CheckFieldIdAt(const ::avro::NodePtr& node, size_t index, int32_t field_id,
   ASSERT_EQ(attrs.getAttribute(key), std::make_optional(std::to_string(field_id)));
 }
 
+// Helper function to check if a custom attribute exists for a field name preservation
+void CheckIcebergFieldName(const ::avro::NodePtr& node, size_t index,
+                           const std::string& original_name) {
+  ASSERT_LT(index, node->customAttributes());
+  const auto& attrs = node->customAttributesAt(index);
+  ASSERT_EQ(attrs.getAttribute("iceberg-field-name"), std::make_optional(original_name));
+}
+
 }  // namespace
 
+TEST(SanitizeFieldNameTest, ValidFieldNames) {
+  // Valid field names should remain unchanged
+  EXPECT_EQ(SanitizeFieldName("valid_field"), "valid_field");
+  EXPECT_EQ(SanitizeFieldName("field123"), "field123");
+  EXPECT_EQ(SanitizeFieldName("_private"), "_private");
+  EXPECT_EQ(SanitizeFieldName("CamelCase"), "CamelCase");
+  EXPECT_EQ(SanitizeFieldName("field_with_underscores"), "field_with_underscores");
+}
+
+TEST(SanitizeFieldNameTest, InvalidFieldNames) {
+  // Field names starting with numbers should be prefixed with underscore
+  EXPECT_EQ(SanitizeFieldName("123field"), "_123field");
+  EXPECT_EQ(SanitizeFieldName("0value"), "_0value");
+
+  // Field names with special characters should have them replaced with underscores
+  EXPECT_EQ(SanitizeFieldName("field-name"), "field_name");
+  EXPECT_EQ(SanitizeFieldName("field.name"), "field_name");
+  EXPECT_EQ(SanitizeFieldName("field name"), "field_name");
+  EXPECT_EQ(SanitizeFieldName("field@name"), "field_name");
+  EXPECT_EQ(SanitizeFieldName("field#name"), "field_name");
+
+  // Complex field names with multiple issues
+  EXPECT_EQ(SanitizeFieldName("1field-with.special@chars"), "_1field_with_special_chars");
+  EXPECT_EQ(SanitizeFieldName("user-email"), "user_email");
+  EXPECT_EQ(SanitizeFieldName("价格"),
+            "_______");  // Non-ASCII characters become underscores
+}
+
+TEST(SanitizeFieldNameTest, EdgeCases) {
+  // Empty field name
+  EXPECT_EQ(SanitizeFieldName(""), "_empty");
+
+  // Field name with only special characters
+  EXPECT_EQ(SanitizeFieldName("@#$"), "____");
```

Comment:
Is this PR mostly generated by AI? This is obviously wrong. I'd suggest to read the Java implementation in detail and be responsible of every line of code to avoid wasting time of reviewers.